src/contracts/libraries/Merkle.sol

```diff
@@ -164,4 +163,45 @@ library Merkle {
         //the first node in the layer is the root
         return layer[0];
     }
+
+    /**
+     * @notice this function returns the merkle root of a tree created from a set of leaves using keccak256 as its hash function
+     *  @param leaves the leaves of the merkle tree
+     *  @return The computed Merkle root of the tree.
+     *  @dev This pads to the next power of 2. very inefficient! just for POC
+     */
+    function merkleizeKeccak256(bytes32[] memory leaves) internal pure returns (bytes32) {
+        uint256 paddedLength = 2;
+        while(paddedLength < leaves.length) {
+            paddedLength <<= 1;
+        }
+
+        bytes32[] memory paddedLeaves = new bytes32[](paddedLength);
+        for (uint256 i = 0; i < leaves.length; i++) {
+            paddedLeaves[i] = leaves[i];
```

Comment:
we should not pad to the next power of 2 here, we should just pad to the next even number

Reply:
eh actually this is probably fine